### ISSUE TITLE
Painter reconnection every 10 seconds, causing a 1 sec freeze

### ISSUE THREAD

**Issue by Prutheus** [DisplayLink/evdi#57]
On my external connected monitor via
    (a) - dockingstation D3100
    (b) - pluggable thunderbolt3-to-hdmi-connector
I get a ca 1 second long freeze every 10 seconds.

In my journal, everytime the lag occurs, I get these Lines:

```
Nov 10 13:16:20 XPS kernel: evdi: [D] evdi_detect:69 (dev=1) Painter is connected
Nov 10 13:16:20 XPS kernel: evdi: [D] evdi_painter_get_edid_copy:192 (dev=1) 00 ff ff
```

What could cause these freezes?
How can I remove them?

**Comment by displaylink-mlukaszek** [comment]
I suppose this is happening only with DisplayLink device? Please raise it on [DisplayLink Forum](http://displaylink.org/forum/forumdisplay.php?f=29), and attach full logs (details in the sticky post). I'd like to keep GitHub for issues that are stictly evdi-specific and would happen with any client - not only DisplayLink.

**Comment by Prutheus** [comment]
I don't think that this is displaylink-related, because when I disable all displaylink service's i still get the freeze, when not connected to the displaylink-device I get the error in my journal: `Nov 16 21:02:10 XPS kernel: evdi: [D] evdi_detect:72 Painter is disconnected`

**Comment by displaylink-mlukaszek** [comment]
OK, that is a valuable information - thanks. Reopening. Can you post a dmesg file with this situation logged? Machine details would also help in reproduction.

**Comment by Prutheus** [comment]
Machine: Dell XPS 13 9360
OS: ArchLinux Kernel 4.8.4-1

dmesg full with:
`[207448.151307] evdi: [D] evdi_painter_get_edid_copy:192 (dev=1) 00 ff ff
[207458.748421] evdi: [D] evdi_detect:69 (dev=1) Painter is connected
[207458.748427] evdi: [D] evdi_painter_get_edid_copy:192 (dev=1) 00 ff ff
[207469.398956] evdi: [D] evdi_detect:69 (dev=1) Painter is connected
[207469.398967] evdi: [D] evdi_painter_get_edid_copy:192 (dev=1) 00 ff ff
[207479.999252] evdi: [D] evdi_detect:69 (dev=1) Painter is connected
[207479.999262] evdi: [D] evdi_painter_get_edid_copy:192 (dev=1) 00 ff ff
[207490.619499] evdi: [D] evdi_detect:69 (dev=1) Painter is connected
[207490.619508] evdi: [D] evdi_painter_get_edid_copy:192 (dev=1) 00 ff ff
[207501.223146] evdi: [D] evdi_detect:69 (dev=1) Painter is connected
[207501.223158] evdi: [D] evdi_painter_get_edid_copy:192 (dev=1) 00 ff ff
[207511.886803] evdi: [D] evdi_detect:69 (dev=1) Painter is connected
[207511.886808] evdi: [D] evdi_painter_get_edid_copy:192 (dev=1) 00 ff ff
[207522.480187] evdi: [D] evdi_detect:69 (dev=1) Painter is connected
[207522.480191] evdi: [D] evdi_painter_get_edid_copy:192 (dev=1) 00 ff ff
[207533.073916] evdi: [D] evdi_detect:69 (dev=1) Painter is connected
[207533.073921] evdi: [D] evdi_painter_get_edid_copy:192 (dev=1) 00 ff ff
[207543.680938] evdi: [D] evdi_detect:69 (dev=1) Painter is connected
[207543.680945] evdi: [D] evdi_painter_get_edid_copy:192 (dev=1) 00 ff ff
[207554.281043] evdi: [D] evdi_detect:69 (dev=1) Painter is connected
[207554.281054] evdi: [D] evdi_painter_get_edid_copy:192 (dev=1) 00 ff ff
[207564.901608] evdi: [D] evdi_detect:69 (dev=1) Painter is connected
[207564.901617] evdi: [D] evdi_painter_get_edid_copy:192 (dev=1) 00 ff ff
[207575.511890] evdi: [D] evdi_detect:69 (dev=1) Painter is connected
[207575.511903] evdi: [D] evdi_painter_get_edid_copy:192 (dev=1) 00 ff ff
[207586.142166] evdi: [D] evdi_detect:69 (dev=1) Painter is connected
[207586.142178] evdi: [D] evdi_painter_get_edid_copy:192 (dev=1) 00 ff ff
[207596.782587] evdi: [D] evdi_detect:69 (dev=1) Painter is connected
[207596.782594] evdi: [D] evdi_painter_get_edid_copy:192 (dev=1) 00 ff ff
[207607.412999] evdi: [D] evdi_detect:69 (dev=1) Painter is connected
[207607.413010] evdi: [D] evdi_painter_get_edid_copy:192 (dev=1) 00 ff ff
[207618.033478] evdi: [D] evdi_detect:69 (dev=1) Painter is connected
[207618.033488] evdi: [D] evdi_painter_get_edid_copy:192 (dev=1) 00 ff ff
[207628.633445] evdi: [D] evdi_detect:69 (dev=1) Painter is connected
[207628.633455] evdi: [D] evdi_painter_get_edid_copy:192 (dev=1) 00 ff ff`

**Comment by Prutheus** [comment]
Here a more detailed log: http://ix.io/1Ge5

`Nov 19 09:22:07 XPS kernel: evdi: [V] painter_unlock:165 
Nov 19 09:22:07 XPS kernel: evdi: [V] evdi_painter_grabpix_ioctl:556 
Nov 19 09:22:07 XPS kernel: evdi: [V] painter_lock:159 
Nov 19 09:22:07 XPS kernel: evdi: [V] copy_pixels:121 
Nov 19 09:22:07 XPS kernel: evdi: [V] copy_pixels:139 copy rect 0,0-1920,1080
Nov 19 09:22:07 XPS kernel: evdi: [V] painter_unlock:165 
Nov 19 09:22:07 XPS kernel: evdi: [V] painter_lock:159 
Nov 19 09:22:07 XPS kernel: evdi: [V] painter_unlock:165 
Nov 19 09:22:07 XPS kernel: evdi: [V] evdi_user_framebuffer_dirty:232 
Nov 19 09:22:07 XPS kernel: evdi: [V] evdi_handle_damage:87 
Nov 19 09:22:07 XPS kernel: evdi: [V] painter_lock:159 
Nov 19 09:22:07 XPS kernel: evdi: [V] painter_unlock:165 
Nov 19 09:22:07 XPS kernel: evdi: [V] painter_lock:159 
Nov 19 09:22:07 XPS kernel: evdi: [V] evdi_painter_mark_dirty:333 (dev=1) 0,0-1920,1080
`

**Comment by displaylink-mlukaszek** [comment]
We've just released new 1.3 version for Ubuntu, which also ships the very latest evdi. Could you test whether this resolves the problem? Thanks.

**Comment by displaylink-mlukaszek** [comment]
Closing due to no feedback.
